app/models/postal_address.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+class PostalAddress < ApplicationRecord
+  belongs_to :user
+  validates_presence_of :address_line_1, :city, :postal_code, :country
+
+  def formatted
+    "#{capitalize(address_line_1)} #{capitalize(address_line_2)} #{capitalize(city)}, #{state_or_province} #{postal_code} #{country}"
```

Comment:
I wanted to pull this logic out of the view, but I like your suggestion of using a partial instead. I'll make that change. And since I'm doing any data validation around field inputs, we need a formatter somewhere to make sure the capitalization is correct.

Reply:
>  I'm doing any data validation around field inputs, we need a formatter somewhere to make sure the capitalization is correct

I'm sorry, but I fear I don't understand a word 🙈 If it's too much hassle to explain again, I can also try to read it again tomorrow when I'm more awake though - no worries.